tools/bin/build_report.py

```diff
@@ -0,0 +1,164 @@
+
+
+def get_status_page(connector) -> str:
+    response = requests.get(f'https://dnsgjos7lj2fu.cloudfront.net/tests/summary/{connector}/index.html')
+    if response.status_code == 200:
+        return response.text
+
+
+def parse(page) -> list:
+    history = []
+    for row in re.findall(r'<tr>(.*?)</tr>', page):
+        cols = re.findall(r'<td>(.*?)</td>', row)
+        if not cols or len(cols) != 3:
+            continue
+        history.append({
+            'date': cols[0],
+            'status': re.findall(r' (\S+)</span>', cols[1])[0],
+            'link': re.findall(r'href="(.*?)"', cols[2])[0],
+        })
+    return history
+
+
+def check_connector(connector):
+    status_page = get_status_page(connector)
+
+    # check if connector is tested
+    if not status_page:
+        NO_TESTS.append(connector)
+        print('F', end='', flush=True)
+        return
+
+    print('.', end='', flush=True)
+
+    if connector.startswith('source'):
+        TESTED_SOURCE.append(connector)
+    elif connector.startswith('destination'):
+        TESTED_DESTINATION.append(connector)
+
+    # order: recent values goes first
+    history = parse(status_page)
+    # order: recent values goes last
+    short_status = ''.join(['.' if build['status'] == 'success' else 'F' for build in history[::-1]])  # ex: F..F..FF
+
+    # check latest build status
+    last_build = history[0]
+    if last_build['status'] == 'success':
+        if connector.startswith('source'):
+            SUCCESS_SOURCE.append(connector)
+        elif connector.startswith('destination'):
+            SUCCESS_DESTINATION.append(connector)
+    else:
+        failed_today = [connector, short_status, last_build['link']]
+
+        if len(history) > 1 and history[1]['status'] != 'success':
+            FAILED_2_LAST.append(failed_today)
+            return
+
+        FAILED_LAST.append(failed_today)
+
+
+def failed_report(failed_report) -> str:
+    max_name_len = max([len(connector[0]) for connector in failed_report])
+    max_status_len = max(len(connector[1]) for connector in failed_report)
+    for connector in failed_report:
+        connector[0] = connector[0].ljust(max_name_len, ' ')
+        connector[1] = connector[1].rjust(max_status_len, ' ')
+    return '\n'.join([' '.join(connector) for connector in failed_report])
+
+
+def create_report(connectors) -> str:
+
+    sources_len = len([name for name in connectors if name.startswith('source')])
+    destinations_len = len([name for name in connectors if name.startswith('destination')])
+
+    report = f"""
+CONNECTORS:   total: {len(connectors)}
+Sources:      total: {sources_len} / tested: {len(TESTED_SOURCE)} / success: {len(SUCCESS_SOURCE)} ({round(len(SUCCESS_SOURCE)/sources_len*100, 1)}%)
+Destinations: total: {destinations_len} / tested: {len(TESTED_DESTINATION)} / success: {len(SUCCESS_DESTINATION)} ({round(len(SUCCESS_DESTINATION)/destinations_len*100, 1)}%)
+
+"""
+    if FAILED_LAST:
```

Comment:
do you think including both last and last2 is useful instead of just last2? What action should a developer take if a connector failed once but not twice?